action.yml

```diff
@@ -0,0 +1,50 @@
+name: Sync Salmon Directory
+description: Syncs a directory to a Google Cloud bucket using rclone.
+author: 'deephaven'
+inputs:
+  source:
+    required: true
+    type: string
+    description: 'The source directory to sync.'
+  destination:
+    required: true
+    type: string
+    description: 'The destination directory to sync. Relative to the bucket.'
+  project_number:
+    required: true
+    type: string
+    description: 'The Google Cloud project number.'
+  bucket:
+    required: true
+    type: string
+    description: 'The Google Cloud bucket to sync to.'
+  credentials:
+    required: true
+    type: string
+    description: 'The Google Cloud credentials.'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup rclone
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y rclone=1.53.3-4ubuntu1.22.04.2
+
+        mkdir -p $HOME/.config
+        mkdir -p $HOME/.config/rclone
+
+        cat << EOF > $HOME/.config/rclone/rclone.conf
+        [docs]
+        type = google cloud storage
+        service_account_file = $HOME/credentials.json
+        project_number = ${{ inputs.project_number }}
+        bucket_policy_only = true
+        EOF
+        
+        echo ${{ inputs.credentials }} | base64 -d > $HOME/credentials.json
```

Comment:
I think this might expose the credentials to the actions log. Maybe github is able to censor it if they're secrets, but not sure in this case since it's an input. Doesn't look like GitHub recognizes GCP keys automatically

You can use `::add-mask::` to ensure this is hidden I believe. https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#redacting-secrets-from-workflow-run-logs

Reply:
There's also OIDC which we might be able to use instead? Not sure how this works. https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect

Reply:
I'm not sure when this would be exposed? Doesn't seem to be in the actions we have using it already. But I'll look into the masking, can't hurt.

Reply:
Looks like as a reusable workflow w/ inherited secrets this redacts since it references `secrets.XYZ` at least.

This is from your fork workflow. The main thing to point out here is the actions log prints the steps. So GitHub redacts the value since it is from `secrets.XYZ`.

![image](https://github.com/user-attachments/assets/ec670df3-e41e-4f57-be2a-b4b98ae401d2)

I would still want to verify if it's a reusable action (where it's just a step in a job instead of a standalone job that needs artifact upload/download) that it redacts this. I'd think it is fine since there are examples on GitHub docs about passing secrets to actions as inputs.

Reply:
![image](https://github.com/user-attachments/assets/512f9243-3636-4e62-a277-99f6c1d912bb)
here it is redacted using this action